src/main/java/io/github/bobocodebreskul/context/support/BeanDefinitionReaderUtils.java

```diff
@@ -35,32 +38,43 @@ public static String generateBeanName(BeanDefinition beanDefinition,
       log.error("Failed to generate bean name for nullable bean definition");
       return "Null bean definition specified";
     });
-    Objects.requireNonNull(beanDefinition.getBeanClass(), () -> {
-      log.error("Failed to generate bean name for nullable bean class");
-      return "Bean class has not been specified";
-    });
+    validateBeanClassNonNull(beanDefinition.getBeanClass());
+    validateBeanClassName(beanDefinition, registry);
```

Comment:
I think we do not need to send the registry here, and maybe better to do validation on the reader level, because it couples us with implementation which says we required to have some registry

Reply:
I assumed that any implementation will require bean definition registry, isn't?
Otherwise we need to refactor everything :(